### PR TITLE
VACMS-6352: Upgrade drupal to 9.3 (including deps)

### PR DESCRIPTION
**Pull request by cweagans**
## Description

Relates to #6352 .

## Testing done

* Unit tests ran (there were a couple failing, but they looked related to my local env, so submitting PR anyway to run them on Tugboat)
* Confirmed phpstan reported deprecated code

## QA steps

As user _uid_ with _user_role_
1. Do this
2. Then that
3. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by cweagans** [comment]
Tugboat build is failing with this:

```
TypeError: Argument 5 passed to Drupal\Core\Template\TwigExtension::__const  
  Drupal\Core\Render\Renderer), Object(Drupal\Core\Render\MetadataBubblingUrl  
  Generator), Object(Drupal\Core\Theme\ThemeManager), Object(Drupal\Core\Date  
  time\DateFormatter), Object(Drupal\flag\FlagCountManager))                   
  ruct() must implement interface Drupal\Core\File\FileUrlGeneratorInterface   
  or be null, instance of Drupal\flag\FlagCountManager given in /var/lib/tugb  
  oat/docroot/core/lib/Drupal/Core/Template/TwigExtension.php on line 86 #0 [  
  internal function]: Drupal\Core\Template\TwigExtension->__construct(Object(  
  ateService(Object(Symfony\Component\DependencyInjection\Definition), Array,  
   false, 'flag.twig.count')                                                   
  #3 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  #1 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(1142): ReflectionClass->newInstanceArgs(Array)                     
  #2 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->cre  
  #4 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(1265): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  Get('flag.twig.count', 1, Array, false)                                      
  ray, false)                                                                  
  #5 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(1621): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  ResolveServices(Array, Array)                                                
  ilder.php(1213): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  ResolveServices(Object(Symfony\Component\DependencyInjection\Reference), Ar  
  #6 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  llMethod(Object(Drupal\Core\Template\TwigEnvironment), Array, Array)         
  #7 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->cre  
  ilder.php(1167): Symfony\Component\DependencyInjection\ContainerBuilder->ca  
  ateService(Object(Symfony\Component\DependencyInjection\Definition), Array,  
   false, 'twig')                                                              
  #8 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(1265): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  Get('twig', 1, Array, false)                                                 
  #9 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerBu  
  ilder.php(1213): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  ResolveServices(Object(Symfony\Component\DependencyInjection\Reference), Ar  
  ray, false)                                                                  
  oResolveServices(Array, Array, false)                                        
  #11 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  #10 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(1115): Symfony\Component\DependencyInjection\ContainerBuilder->d  
  uilder.php(1213): Symfony\Component\DependencyInjection\ContainerBuilder->d  
  uilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->cr  
  eateService(Object(Symfony\Component\DependencyInjection\Definition), Array  
  , false, 'twig.commands')                                                    
  #12 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(1265): Symfony\Component\DependencyInjection\ContainerBuilder->d  
  oGet('twig.commands', 3, Array, false)                                       
  #13 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  #16 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->cr  
  eateService(Object(Symfony\Component\DependencyInjection\Definition), Array  
  , true, 'drush.command.s...')                                                
  #17 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(558): Symfony\Component\DependencyInjection\ContainerBuilder->do  
  Get('drush.command.s...', 1)                                                 
  #18 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/DrupalBoot8.php(27  
  6): Symfony\Component\DependencyInjection\ContainerBuilder->get('drush.comm  
  and.s...')                                                                   
  #19 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/DrupalBoot8.php(24  
  oResolveServices(Object(Symfony\Component\DependencyInjection\Reference), A  
  rray, false)                                                                 
  #14 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(1621): Symfony\Component\DependencyInjection\ContainerBuilder->d  
  oResolveServices(Array, Array)                                               
  #15 /var/lib/tugboat/docroot/vendor/symfony/dependency-injection/ContainerB  
  uilder.php(1167): Symfony\Component\DependencyInjection\ContainerBuilder->c  
  allMethod(Object(Drush\Command\ServiceCommandlist), Array, Array)            
  hp(432): Drush\Boot\BootstrapManager->doBootstrap(5, 5, Object(Consolidatio  
  n\AnnotatedCommand\AnnotationData))                                          
  #22 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/BootstrapManager.p  
  hp(384): Drush\Boot\BootstrapManager->bootstrapToPhaseIndex(5, Object(Conso  
  2): Drush\Boot\DrupalBoot8->addDrupalModuleDrushCommands(Object(Drush\Boot\  
  BootstrapManager))                                                           
  #20 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/BootstrapManager.p  
  hp(291): Drush\Boot\DrupalBoot8->bootstrapDrupalFull(Object(Drush\Boot\Boot  
  strapManager), Object(Consolidation\AnnotatedCommand\AnnotationData))        
  #21 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/BootstrapManager.p  
  lidation\AnnotatedCommand\AnnotationData))                                   
  #23 /var/lib/tugboat/docroot/vendor/drush/drush/src/Boot/BootstrapHook.php(  
  #25 /var/lib/tugboat/docroot/vendor/consolidation/annotated-command/src/Hoo  
  ks/Dispatchers/InitializeHookDispatcher.php(36): Consolidation\AnnotatedCom  
  32): Drush\Boot\BootstrapManager->bootstrapToPhase('full', Object(Consolida  
  tion\AnnotatedCommand\AnnotationData))                                       
  #24 /var/lib/tugboat/docroot/vendor/consolidation/annotated-command/src/Hoo  
  ks/Dispatchers/InitializeHookDispatcher.php(44): Drush\Boot\BootstrapHook->  
  initialize(Object(Drush\Symfony\DrushArgvInput), Object(Consolidation\Annot  
  atedCommand\AnnotationData))                                                 
  mand\Hooks\Dispatchers\InitializeHookDispatcher->callInitializeHook(Object(  
  #26 /var/lib/tugboat/docroot/vendor/consolidation/annotated-command/src/Hoo  
  ks/Dispatchers/InitializeHookDispatcher.php(29): Consolidation\AnnotatedCom  
  mand\Hooks\Dispatchers\InitializeHookDispatcher->doInitializeHook(Object(Dr  
  ush\Boot\BootstrapHook), Object(Drush\Symfony\DrushArgvInput), Object(Conso  
  lidation\AnnotatedCommand\AnnotationData))                                   
  Drush\Boot\BootstrapHook), Object(Drush\Symfony\DrushArgvInput), Object(Con  
  solidation\AnnotatedCommand\AnnotationData))                                 
  #27 /var/lib/tugboat/docroot/vendor/consolidation/annotated-command/src/Com  
  mandProcessor.php(145): Consolidation\AnnotatedCommand\Hooks\Dispatchers\In  
  itializeHookDispatcher->initialize(Object(Drush\Symfony\DrushArgvInput), Ob  
  ject(Consolidation\AnnotatedCommand\AnnotationData))                         
  #28 /var/lib/tugboat/docroot/vendor/consolidation/annotated-command/src/Ann  
  otatedCommand.php(335): Consolidation\AnnotatedCommand\CommandProcessor->in  
  itializeHook(Object(Drush\Symfony\DrushArgvInput), Array, Object(Consolidat  
  ion\AnnotatedCommand\AnnotationData))                                        
  #29 /var/lib/tugboat/docroot/vendor/symfony/console/Command/Command.php(221  
  ): Consolidation\AnnotatedCommand\AnnotatedCommand->initialize(Object(Drush  
  \Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOu  
  tput))                                                                       
  #30 /var/lib/tugboat/docroot/vendor/symfony/console/Application.php(1027):   
  Symfony\Component\Console\Command\Command->run(Object(Drush\Symfony\DrushAr  
  gvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))            
  ct(Symfony\Component\Console\Output\ConsoleOutput))                          
  #32 /var/lib/tugboat/docroot/vendor/symfony/console/Application.php(149): S  
  ymfony\Component\Console\Application->doRun(Object(Drush\Symfony\DrushArgvI  
  nput), Object(Symfony\Component\Console\Output\ConsoleOutput))               
  #33 /var/lib/tugboat/docroot/vendor/drush/drush/src/Runtime/Runtime.php(121  
  #31 /var/lib/tugboat/docroot/vendor/symfony/console/Application.php(273): S  
  ymfony\Component\Console\Application->doRunCommand(Object(Consolidation\Ann  
  otatedCommand\AnnotatedCommand), Object(Drush\Symfony\DrushArgvInput), Obje  
   require('/var/lib/tugboa...')                                               
  #37 phar:///usr/local/bin/drush/bin/drush.php(141): drush_main()             
  #34 /var/lib/tugboat/docroot/vendor/drush/drush/src/Runtime/Runtime.php(51)  
  : Drush\Runtime\Runtime->doRun(Array, Object(Symfony\Component\Console\Outp  
  ut\ConsoleOutput))                                                           
  #35 /var/lib/tugboat/docroot/vendor/drush/drush/drush.php(72): Drush\Runtim  
  e\Runtime->run(Array)                                                        
  #36 /var/lib/tugboat/docroot/vendor/drush/drush/includes/preflight.inc(18):  
  ): Symfony\Component\Console\Application->run(Object(Drush\Symfony\DrushArg  
  vInput), Object(Symfony\Component\Console\Output\ConsoleOutput))             
                                                                               

  #38 /usr/local/bin/drush(10): require('phar:///usr/loc...')                  
  #39 {main}                                                                   
   [warning] Drush command terminated abnormally.
```

Not sure what's going on there, but I'll get it figured out.

File: composer.json [hunk 1]
```diff
@@ -394,9 +392,6 @@
             "drupal/entity_reference_revisions": {
                 "2984027 Incorrect manipulation of default revision flag": "https://www.drupal.org/files/issues/2020-09-03/entity_reference_revisions-default_revision-2984027-10.patch"
             },
-            "drupal/flag": {
-                "2957019 Failed to start the session because headers have already been sent": "https://www.drupal.org/files/issues/2020-01-20/2957019-51.patch"
```
Comment (swirtSJW) [inline composer.json:398]: This is a long issue
https://www.drupal.org/project/flag/issues/2957019
following it to here 
https://www.drupal.org/project/flag/issues/3229964/
Makes it read as though this patch is no longer needed.  

Just logging the chain here for record.

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 2]
```diff
@@ -110,7 +110,10 @@ function _va_gov_rename_vocab_term($oldname, $newname, $vocabulary_name = NULL)
     ':oldname' => $oldname,
     ':newname' => $newname,
   ];
-  $terms = taxonomy_term_load_multiple_by_name($oldname, $vocabulary_name);
+  $terms = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadByProperties([
+    'name' => $oldname,
+    'vid' => $vocabulary_name,
+  ]);
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.install:116]: This looks good.  No need to fully test this as part of QA as the failure will turn up the next time we need to use this in hook_update_N

File: docroot/modules/custom/va_gov_graphql/va_gov_graphql.module [hunk 3]
```diff
@@ -26,3 +27,35 @@ function va_gov_graphql_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Get a list of menus excluding Drupal-defined menus.
+ *
+ * This is a replacement for menu_ui_get_menus(), which was deprecated in
+ * Drupal 9.3.0.
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_graphql/va_gov_graphql.module:35]: docblock does not declare the @return

Not sure how this passed phpcs.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
These look good.  Thank you for digging into it.  

 There is only one change requested.  Everything else was just breadcrumbs

**Comment by cweagans** [comment]
Closed in favor of https://github.com/department-of-veterans-affairs/va.gov-cms/pull/8076

**Pull request closed without merging**